httpclient5/src/main/java/org/apache/hc/client5/http/async/methods/DecompressingAsyncEntityConsumer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+package org.apache.hc.client5.http.async.methods;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import org.apache.hc.client5.http.entity.compress.ContentCodecRegistry;
+import org.apache.hc.core5.concurrent.FutureCallback;
+import org.apache.hc.core5.http.EntityDetails;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpException;
+import org.apache.hc.core5.http.nio.AsyncEntityConsumer;
+import org.apache.hc.core5.http.nio.CapacityChannel;
+
+/**
+ * An {@link AsyncEntityConsumer} implementation that consumes and decompresses
+ * an HTTP response body to produce a {@link String}, based on the value of the
+ * {@code Content-Encoding} header. Supports all built-in and Commons Compress
+ * codecs registered in {@link ContentCodecRegistry}.
+ *
+ * <p>
+ * This class buffers the entire response entity content into memory before
+ * applying decompression. Not suitable for large payloads.
+ * </p>
+ *
+ * @since 5.6
+ */
+public class DecompressingAsyncEntityConsumer<T> implements AsyncEntityConsumer<T> {
```

Comment:
@arturobernalg This class looks completely redundant. I pulled your change-set, removed the class, ran the tests and the example and they all worked for me.